install/autodetect.go

```diff
@@ -69,12 +102,22 @@ func (k *K8sInstaller) autodetectAndValidate(ctx context.Context) error {
 		k.Log("🔮 Auto-detected Kubernetes kind: %s", f.Kind)
 	}
 
+	return nil
+}
+
+func (k *K8sInstaller) autodetectAndValidate(ctx context.Context) error {
+	if err := k.autodetect(ctx); err != nil {
+		return err
+	}
+
+	f := k.flavor
```

Comment:
Why take this copy? Doesn't look like we mutate `f`

Reply:
The initial idea was to not change too much about the current code. Agree that the copy seems unnecessary, will change.